src/time.c

```diff
@@ -46,6 +46,61 @@ rcutils_time_point_value_as_nanoseconds_string(
   return RCUTILS_RET_OK;
 }
 
+rcutils_ret_t
+rcutils_time_point_value_as_date_string(
+  const rcutils_time_point_value_t * time_point,
+  char * str,
+  size_t str_size)
+{
+  RCUTILS_CHECK_ARGUMENT_FOR_NULL(time_point, RCUTILS_RET_INVALID_ARGUMENT);
+  RCUTILS_CHECK_ARGUMENT_FOR_NULL(str, RCUTILS_RET_INVALID_ARGUMENT);
+  if (0 == str_size) {
+    return RCUTILS_RET_OK;
+  }
+  // best to abs it to avoid issues with negative values in C89, see:
+  //   https://stackoverflow.com/a/3604984/671658
+  uint64_t abs_time_point = (uint64_t)llabs(*time_point);
+  // break into two parts to avoid floating point error
+  uint64_t seconds = abs_time_point / (1000u * 1000u * 1000u);
+  uint64_t nanoseconds = abs_time_point % (1000u * 1000u * 1000u);
+  // Make sure the buffer is large enough to hold the largest possible uint64_t
+  char nanoseconds_str[21];
```

Comment:
why we need to allocate largest possible `uint64_t` in this case? should be less than 10?

```suggestion
  char nanoseconds_str[10];
```

Reply:
The largest number a uint64 can be (UINT64_MAX) is: 18446744073709551615 .  That is 20 characters long, plus one character for a \0.  While the nanoseconds value probably shouldn't be this long, I think we should be conservative and handle it.  In any case, this is stack memory, so it is extremely cheap and will be freed automatically at the end of the function.